changelog/670.feature.rst

```diff
@@ -0,0 +1,14 @@
+|commands| Add support for injected parameters autocompletion
+	Document :class:`.ext.commands.Injection`
+
+	:meth:`.ext.commands.Injection.register` now returns :class:`.ext.commands.Injection`
+
+	:func:`.ext.commands.register_injection` now returns :class:`.ext.commands.Injection`

```

Comment:
I'd argue these two return type changes should be marked as breaking, just in case. @onerandomusername thoughts?

Reply:
I guess thoss return values weren't meant for public use previously (i.e., not a part of a public api) and thus this change shouldn't be considered breaking
There also was a related conversation on discord: https://discord.com/channels/808030843078836254/913779868985090089/1002217259374227467

Reply:
At least changing `register_injection` is breaking, but `Injection.register` was typed as returning `Any`. IMO if moving `register_injection` to a breaking changelog entry, we should move `Injection.register` as well.

Reply:
So, should we consider these changes breaking?

Reply:
We should mark them as breaking, I'd rather have something labeled as breaking without it actually breaking anything, than the other way around.
The runtime impact could be avoided by making `Injection` generic and adding a `__call__` that passes calls through to `self.function`, such that this would basically end up only being a breaking change in terms of typing.

Reply:
Im too lazy to reference commits, but done